### PR TITLE
docs: add missing throw descriptions in JSDocs

### PR DESCRIPTION
**Pull request by xDivisionByZerox**
Add missing and updated `@throw` descriptions in the JSDocs for the following functions:

| Updated               | Added                 |
| --------------------- | --------------------- |
| faker.number.int      | faker.number.int      |
| faker.number.float    | faker.number.float    |
| faker.number.binary   | faker.number.binary   |
| faker.number.octal    | faker.number.octal    |
| faker.number.hex      | faker.number.hex      |
| faker.number.bigInt   |                       |
| faker.datatype.number | faker.datatype.number |
|                       | faker.datatype.float  |

Additionally, to make sure that those statements are actually true, I adjusted or extended the test for those functions as well.

### CODE REVIEW

**Comment by ST-DDT** [comment]
This is somewhat blocked by #2444

- #2444

**Review by ST-DDT — DISMISSED** [review]

**Comment by xDivisionByZerox** [comment]
> This is somewhat blocked by https://github.com/faker-js/faker/pull/2444

Should we add https://github.com/faker-js/faker/labels/s:%20on%20hold?

**Comment by ST-DDT** [comment]
The jsdocs get improved by this regardless, so I don't consider it worth hard blocking.
(It only looks slightly ugly in the api-docs)

File: src/modules/number/index.ts [hunk 1]
```diff
@@ -24,7 +24,8 @@ export class NumberModule extends SimpleModuleBase {
    * @param options.min Lower bound for generated number. Defaults to `0`.
    * @param options.max Upper bound for generated number. Defaults to `Number.MAX_SAFE_INTEGER`.
    *
-   * @throws When options define `max < min`.
+   * @throws When `min` is greater than `max`.
+   * @throws When `min` and `max` do not provide a sufficient range to generate integers in between them.
```
Comment (ejcheng) [inline src/modules/number/index.ts]:
```suggestion
   * @throws When `min` and `max` do not form a sufficient range to generate integers from.
```

File: src/modules/number/index.ts [hunk 2]
```diff
@@ -168,7 +172,8 @@ export class NumberModule extends SimpleModuleBase {
    * @param options.min Lower bound for generated number. Defaults to `0`.
    * @param options.max Upper bound for generated number. Defaults to `1`.
    *
-   * @throws When options define `max < min`.
+   * @throws When `min` is greater than `max`.
+   * @throws When `min` and `max` do not provide a sufficient range to generate integers in between them.
```
Comment (ejcheng) [inline src/modules/number/index.ts]:
```suggestion
   * @throws When `min` and `max` do not form a sufficient range to generate integers from.
```

File: src/modules/number/index.ts [hunk 3]
```diff
@@ -217,7 +222,8 @@ export class NumberModule extends SimpleModuleBase {
    * @param options.min Lower bound for generated number. Defaults to `0`.
    * @param options.max Upper bound for generated number. Defaults to `7`.
    *
-   * @throws When options define `max < min`.
+   * @throws When `min` is greater than `max`.
+   * @throws When `min` and `max` do not provide a sufficient range to generate integers in between them.
```
Comment (ejcheng) [inline src/modules/number/index.ts]:
```suggestion
   * @throws When `min` and `max` do not form a sufficient range to generate integers from.
```

File: src/modules/number/index.ts [hunk 4]
```diff
@@ -266,7 +272,8 @@ export class NumberModule extends SimpleModuleBase {
    * @param options.min Lower bound for generated number. Defaults to `0`.
    * @param options.max Upper bound for generated number. Defaults to `15`.
    *
-   * @throws When options define `max < min`.
+   * @throws When `min` is greater than `max`.
+   * @throws When `min` and `max` do not provide a sufficient range to generate integers in between them.
```
Comment (ejcheng) [inline src/modules/number/index.ts]:
```suggestion
   * @throws When `min` and `max` do not form a sufficient range to generate integers from.
```

File: src/modules/number/index.ts [hunk 5]
```diff
@@ -313,7 +320,7 @@ export class NumberModule extends SimpleModuleBase {
    * @param options.min Lower bound for generated bigint. Defaults to `0n`.
    * @param options.max Upper bound for generated bigint. Defaults to `min + 999999999999999n`.
    *
-   * @throws When options define `max < min`.
+   * @throws When `min` is greater to `max`.
```
Comment (ejcheng) [inline src/modules/number/index.ts]:
```suggestion
   * @throws When `min` is greater than `max`.
```

File: src/modules/number/index.ts (hunk 1 above)
Comment (matthewmayer) [reply src/modules/number/index.ts]: How about just "when there are no integers between min and max"

**Comment by ST-DDT** [comment]
This is no longer soft blocked by #2444.

- #2444

File: src/modules/number/index.ts (hunk 1 above)
Comment (xDivisionByZerox) [reply src/modules/number/index.ts]:
```suggestion
   * @throws When there are no integers between `min` and `max`.
```

**Review by matthewmayer — APPROVED** [review]

**Review by ST-DDT — APPROVED** [review]

**Review by ejcheng — APPROVED** [review]

**Pull request merged**